tools/build.js

```diff
@@ -1,31 +1,120 @@
-require('colors');
-const fse = require('fs-extra');
+const { green, cyan, red } = require('chalk');
+const webpack = require('webpack');
 
-const bower = require('./build-bower');
-const lib = require('./build-lib');
-const es = require('./build-es');
-const dist = require('./build-dist');
-const { distRoot, bowerRoot } = require('./constants');
+const path = require('path');
+const fse = require('fs-extra');
+const execa = require('execa');
+const getConfig = require('./dist.webpack.config');
 
 const targets = process.argv.slice(2);
+const stdio = ['pipe', 'pipe', 'inherit'];
+
+const srcRoot = path.join(__dirname, '../src/');
+const distRoot = path.join(__dirname, '../dist/');
+const libRoot = path.join(__dirname, '../lib/');
+const esRoot = path.join(__dirname, '../es/');
+const bowerRoot = path.join(__dirname, '../amd/');
+
+const clean = async dir => fse.existsSync(dir) && fse.remove(dir);
+
+const step = (name, root, fn) => async () => {
+  console.log(cyan('Building: ') + green(name));
+  await clean(root);
+  await fn();
+  console.log(cyan('Built: ') + green(name));
+};
+
 const has = t => !targets.length || targets.includes(t);
 
+/**
+ * Run babel over the src directory and output
+ * compiled common js files to ./lib.
+ */
+const buildLib = step('commonjs modules', libRoot, () =>
+  execa.shell(`npx babel ${srcRoot} --out-dir ${libRoot} --env-name "lib"`, {
```

Comment:
why npx instead of just using a dev dep?

Reply:
it is a dev dep, but this saves me from having to do `node_modules/.bin/node`

Reply:
fair enough i guess